[Bug][Go][Client] Downloaded File Does Not Exist When Returned to Caller

icubbon opened the pull request:
<!-- Enter details of the change here. Include additional tests that have been done, reference to the issue for tracking, etc. -->
Keep the file that gets written out to by the client->decode function

The calling code should be responsible for cleaning up the file returned from the Client. The file is already a `tempFile` so it is cleaned up by the OS eventually but can be cleaned up sooner.

Reverts #16303 
Fixes https://github.com/OpenAPITools/openapi-generator/issues/16953

<!-- Please check the completed items below -->
### PR checklist
 
- [x] Read the [contribution guidelines](https://github.com/openapitools/openapi-generator/blob/master/CONTRIBUTING.md).
- [x] Pull Request title clearly describes the work in the pull request and Pull Request description provides details about how to validate the work. Missing information here may result in delayed response from the community.
- [x] Run the following to [build the project](https://github.com/OpenAPITools/openapi-generator#14---build-projects) and update samples:
  ```
  ./mvnw clean package 
  ./bin/generate-samples.sh ./bin/configs/*.yaml
  ./bin/utils/export_docs_generators.sh
  ``` 
  Commit all changed files. 
  This is important, as CI jobs will verify _all_ generator outputs of your HEAD commit as it would merge with master. 
  These must match the expectations made by your contribution. 
  You may regenerate an individual generator by passing the relevant config(s) as an argument to the script, for example `./bin/generate-samples.sh bin/configs/java*`. 
  For Windows users, please run the script in [Git BASH](https://gitforwindows.org/).
- [x] File the PR against the [correct branch](https://github.com/OpenAPITools/openapi-generator/wiki/Git-Branches): `master` (upcoming 7.1.0 minor release - breaking changes with fallbacks), `8.0.x` (breaking changes without fallbacks)
- [x] If your PR is targeting a particular programming language, @mention the [technical committee](https://github.com/openapitools/openapi-generator/#62---openapi-generator-technical-committee) members, so they are more likely to review the pull request.

@wing328

wing328 commented:
cc @antihax (2017/11) @grokify (2018/07) @kemokemo (2018/09) @jirikuncar (2021/01) @ph4r5h4d (2021/04) @lwj5 (2023/04)

wing328 commented:
@icubbon thanks for the fix. can you please merge the latest master into your branch as I've just merged https://github.com/OpenAPITools/openapi-generator/pull/16956 so that you can leverage the Echo API client to test this fix and get it covered moving forward?

Can you please add a test similar to https://github.com/OpenAPITools/openapi-generator/blob/master/samples/client/echo_api/python/test/test_manual.py#L110C29-L110C29?

To run echo server locally, please refer to https://github.com/OpenAPITools/openapi-generator/wiki/Integration-Tests#echo-server

icubbon commented:
@wing328 It looks like there is already a test that is being skipped that would catch this but it was being skipped. I removed the `t.Skip` call and fixed the compilation errors in that package. The tests won't run unless `go get github.com/stretchr/testify/assert` is run to add `testify` to the go.mod file.

But that test now runs. I did confirm that it catches the problem and confirms that it is fixed. 

https://github.com/OpenAPITools/openapi-generator/blob/4402635d4934d62a5c464e43d5cf79478a4128cd/samples/client/echo_api/go/test/api_body_test.go#L25
